doc/user/metrics/service_monitor.md

```diff
@@ -0,0 +1,28 @@
+## Using the ServiceMonitor prometheus-operator CRD
+
+[prometheus-operator][prom-operator] an operator that creates, configures and manages Prometheus clusters atop Kubernetes.
+
+`ServiceMonitor` is a [CR][cr] of the prometheus-operator, which discovers the `Endpoints` in `Service` objects and configures Prometheus to monitor those `Pod`s. The `GenerateServiceMonitor` takes a `Service` object and generates a `ServiceMonitor` resource based on it.
+To add `Service` target discovery of your created monitoring `Service`s you can use the `metrics.CreateServiceMonitor()` helper function, which accepts the newly created `Service`.
```

Comment:
We should probably add some context about when it might be useful to use a service monitor. Is this something users are supposed to create to monitor their operator, resources created by their operator, or both?

Reply:
As ServiceMonitor is strictly a prom-operator concept, I linked to their architecture docs on this. I think those who are interested in creating the object probably already know why they want it? The general use case is most likely when you have a lot of Services and use the operator.